lib/OpenQA/Worker/Jobs.pm

```diff
@@ -511,6 +512,18 @@ sub upload_status(;$) {
     }
 }
 
+sub optimize_image {
+    my ($image) = @_;
+
+    if (which('optipng')) {
+        print("optipng $image\n") if $verbose;
+        # be careful not to be too eager optimizing, this needs to be quick
+        # or we will be considered a dead worker
+        system('optipng', '-quiet', '-o2', $image);
```

Comment:
yup. the preserve only touches mtime - as we only upload the content not the timestamp, it doesn't matter - and it reduces the code size 10 characters ;)